pallets/parachain-system/src/lib.rs

```diff
@@ -544,6 +582,19 @@ pub mod pallet {
 		Unauthorized,
 	}
 
+	/// Latest parachain head data included in the relay chain.
+	///
+	/// This value is optional since it requires extra relay-chain state proof.
+	#[pallet::storage]
+	pub(super) type ParaHead<T: Config> = StorageValue<_, relay_chain::HeadData, OptionQuery>;
```

Comment:
This shouldn't need to hit storage - the latest included head should be read out of the state proof in the paras-inherent.

Reply:
[ffe3dcb](https://github.com/paritytech/cumulus/pull/2438/commits/ffe3dcb5fc491350887323b1777341efc90e23f5)